src/cpu/aarch64/acl_reorder.hpp

```diff
@@ -118,26 +140,71 @@ struct acl_reorder_fwd_t : public primitive_t {
 
             auto src_tag = memory_desc_matches_one_of_tag(
                     *src_md, format_tag::ab, format_tag::ba, format_tag::cdba);
-            ACL_CHECK_SUPPORT(format_tag::undef == src_tag,
+            VDISPATCH_REORDER_IC(format_tag::undef != src_tag,
                     "Only ab, ba or cdba source formats supported");
 
+            // We need this check as ACL validation is currently broken.
+            // We can remove this once the fix is released and we bump the
+            // min supported version to ACL v53.
             auto dst_tag = memory_desc_matches_one_of_tag(*dst_md,
                     format_tag::BA8b4a, format_tag::BA4b4a, format_tag::Ab4a,
                     format_tag::Ab8a, format_tag::Acdb8a, format_tag::Acdb4a);
-            ACL_CHECK_SUPPORT(format_tag::undef == dst_tag,
+            VDISPATCH_REORDER_IC(format_tag::undef != dst_tag,
                     "Only Ab4a/Ab8a, BA8b4a/BA4b4a and Acdb8a/Acdb4a "
                     "destination formats supported");
 
-            if (dst_tag == format_tag::BA4b4a || dst_tag == format_tag::Acdb4a
-                    || dst_tag == format_tag::Ab4a) {
-                _pd->app_.dst_wf = arm_compute::WeightFormat::OHWIo4;
-            } else if (mayiuse(sve_256)
-                    && (dst_tag == format_tag::BA8b4a
-                            || dst_tag == format_tag::Acdb8a
-                            || dst_tag == format_tag::Ab8a)) {
-                _pd->app_.dst_wf = arm_compute::WeightFormat::OHWIo8;
+            auto &transpose = _pd->app_.transpose;
+            auto &dst_blocking = dst_md->format_desc.blocking;
+
+            VDISPATCH_REORDER_IC(src_md->ndims == dst_md->ndims,
+                    "Number of dimensions in src and dst do not match");
+            VDISPATCH_REORDER_IC((dst_md->ndims == 2 || dst_md->ndims == 4),
+                    "ACL only supports 2D and 4D reorders");
+            // Check if a transpose is needed during the reorder
+            if (src_md->ndims == 4) {
+                if (memory_desc_matches_tag(
+                            *src_md, dnnl::impl::format_tag::cdba)
+                        && (memory_desc_matches_tag(
+                                    *dst_md, dnnl::impl::format_tag::Acdb4a)
+                                || memory_desc_matches_tag(*dst_md,
+                                        dnnl::impl::format_tag::Acdb8a))) {
+                    transpose = true;
+                } else {
+                    return status::unimplemented;
```

Comment:
Given that `VDISPATCH_REORDER_IC` was introduced, it's a good practice to rely on it at other spots as well.